docs/proposals/rollout/README.md

```diff
@@ -0,0 +1,308 @@
+
+#### Story #2: Batch Release with Partition
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: CloneSet
+metadata:
+  name: rollout-demo
+spec:
+  replicas: 10
+  updateStrategy:
+    partition: 4
+    maxUnavailable: 2
+  template:
+    metadata:
+      labels:
+        app: rollout-demo
+    spec:
+      containers:
+        - name: main
+          image: nginx:v2 # upgrade rollout-demo from V1 -> V2
+  selector:
+    matchLabels:
+      app: rollout-demo
+---
+apiVersion: karmada.io/v1alpha2
+kind: ResourceBinding
+metadata:
+  name: rollout-demo-cloneset
+spec:
+  clusters: # scheduled to 2 clusters
+    - name: cluster-1
+      replicas: 3
+    - name: cluster-2
+      replicas: 7
+```
+
+
+We expect that the CloneSet cloud be paused when the number of Pods with the new image reaches 4, and the rollout process will be consistent with the semantics in a single-cluster setting. 
+During rolling processing, the number of unavailable Pods will not exceed 2.
+
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+- May further reduce the efficiency of distribution synchronization of workloads.
+
+## Architecture
+The primary approach involves using feedback from the StatusController on single-cluster workload statuses to derive the next rollout configurations. This ensures that throughout the Rollout process, rollout configurations such as `Partition`, `MaxUnavailable`, and `MaxSurge` maintain their original native meanings, and that the federated Rollout behavior aligns with single-cluster  practices.
+
+![architecture](rollout-mechanism-arch.png)
+
+## Changes required
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Code Changes required
+- New Methods in Resource Interpreter
+```golang
+package resourceinterpreter
+
+type ResourceInterpreter interface {
+	// Get & Set for Partition
+	GetPartition(object *unstructured.Unstructured) (*intstr.IntOrString, error)
+	RevisePartition(object *unstructured.Unstructured, partition *intstr.IntOrString) (*unstructured.Unstructured, error)
+
+	// Get & Set for MaxSurge
+	GetMaxSurge(object *unstructured.Unstructured) (*intstr.IntOrString, error)
+	ReviseMaxSurge(object *unstructured.Unstructured, maxSurge *intstr.IntOrString) (*unstructured.Unstructured, error)
+
+	// Get & Set for MaxUnavailble
+	GetMaxUnavailable(object *unstructured.Unstructured) (*intstr.IntOrString, err error)
+	ReviseMaxUnavailable(object *unstructured.Unstructured, maxUnavailable *intstr.IntOrString) (*unstructured.Unstructured, error)
+
+	RolloutInterperter
```

Comment:
As we talked at the community meeting, the rollout algorithm may not be an interpreter, it can leverage the interpreter to get/set things like `MaxUnavailable`.